exporter/azure_monitor/azure_monitor.go

```diff
@@ -37,17 +38,87 @@ var _ trace.Exporter = (*AzureTraceExporter)(nil)
+		Time : getCurrentTime(),
+	}
+
+	if sd.ParentSpanID.String() == "" { 
+		// TODO: Add parent span details if any
+		fmt.Println("ADD PARENT DETAILS")
+		//envelope.Tags["ai.operation.parentId"] = 
+	}
+	if sd.SpanKind == trace.SpanKindServer {
+		// TODO: Add for server case
+		fmt.Println("ADD SERVER CASE")
+		envelope.Name = "Microsoft.ApplicationInsights.Request"
+	} else {
+		envelope.Name = "Microsoft.ApplicationInsights.RemoteDependency"
+		currentData := common.RemoteDependency{
+			Name : sd.Name,
+			Id : "|" + sd.SpanContext.TraceID.String() + "." + sd.SpanID.String() + ".",
+			ResultCode : "0", // TODO: Out of scope for now
+			Duration : timeStampToDuration(sd.EndTime.Sub(sd.StartTime)),
+			Success : true,
+			Ver : 2,
+		}
+		if sd.SpanKind == trace.SpanKindClient {
+			// TODO: Add for client case
+			fmt.Println("ADD CLIENT CASE")
+		} else {
+			currentData.Type = "INPROC" 
+		}
+		envelope.DataToSend = common.Data {
+			BaseData : currentData,
+			BaseType : "RemoteDependencyData",
+		}
 	}
-	envelope.Name = "Microsoft.ApplicationInsights.RemoteDependency"
 
 	transporter := common.Transporter{ 
 		EnvelopeData: envelope,
 	}
-	transporter.Transmit(&exporter.options, &envelope)
+	transporter.Transmit(&exporter.Options, &envelope)
 
 	fmt.Printf("Name: %s\nTraceID: %x\nSpanID: %x\nParentSpanID: %x\nStartTime: %s\nEndTime: %s\nAnnotations: %+v\n\n",
 		sd.Name, sd.TraceID, sd.SpanID, sd.ParentSpanID, sd.StartTime, sd.EndTime, sd.Annotations)
 }
+
+/* Generates the current time stamp and properly formats to a string.
+	@return time stamp
+*/
+func getCurrentTime() string {
+	t := time.Now()
+	formattedTime := t.Format("2006-01-02T15:04:05.000000Z")
+	return formattedTime
+}
+
+/* Calculates number of days, hours, minutes, seconds, and milliseconds in a
+	time duration. Then it properly formats into a string.
+	@param t Time Duration
+	@return formatted string 
+*/
+func timeStampToDuration(t time.Duration) (string) { 
+	nanoSeconds := t.Nanoseconds()
+	milliseconds, remainder :=  divMod(nanoSeconds, 1000000)
+	seconds, remainder :=       divMod(remainder, 1000)
+	minutes, remainder :=       divMod(remainder, 60)
+	hours, remainder :=         divMod(remainder, 60)
+	days, remainder :=          divMod(remainder, 24)
+	
+	formattedDays:=          fmt.Sprintf("%01d", days)
+	formattedHours:=         fmt.Sprintf("%02d", hours)
+	formattedMinutes :=      fmt.Sprintf("%02d", minutes)
+	formattedSeconds :=      fmt.Sprintf("%02d", seconds)
+	formattedMilliseconds := fmt.Sprintf("%03d", milliseconds)
+
+	return formattedDays + "." + formattedHours + ":" + formattedMinutes + ":" + formattedSeconds + "."+ formattedMilliseconds
+}
+
+/* Performs division and returns both quotient and remainder. */
+func divMod(numerator, denominator int64) (quotient, remainder int64) {
+    quotient = numerator / denominator // integer division, decimals are truncated
+    remainder = numerator % denominator
+    return

```

Comment:
curious, what would happen if we remove this line?

Reply:
I get a compilation error if I don't add a return statement. It does look a bit ugly not sure how else to avoid it.

Reply:
Wait, actually let me try to see if there is a way. There might be.

Reply:
This works. I think I will change to the one liner then.
```suggestion
     return (numerator / denominator), (numerator % denominator)
```